src/main/java/com/sap/cap/cds/intellij/lsp/ServerDescriptor.java

```diff
@@ -9,86 +9,75 @@
 import com.intellij.platform.lsp.api.customization.LspFormattingSupport;
 import com.intellij.util.io.BaseOutputReader;
 import com.sap.cap.cds.intellij.FileType;
-import com.sap.cap.cds.intellij.util.Logger;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
-import org.apache.maven.artifact.versioning.ComparableVersion;
 
-import java.nio.file.Files;
+import java.io.BufferedReader;
+import java.io.IOException;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 
 import static com.intellij.util.PathUtil.getJarPathForClass;
-import static com.sap.cap.cds.intellij.util.JsonUtil.getPropertyAtPath;
+import static java.nio.charset.Charset.defaultCharset;
 import static java.nio.charset.StandardCharsets.UTF_8;
 
 public class ServerDescriptor extends ProjectWideLspServerDescriptor {
 
     private static final String RELATIVE_SERVER_PATH = "cds-lsp/node_modules/@sap/cds-lsp/dist/main.js";
-    private static final String RELATIVE_SERVER_PKG_PATH = "cds-lsp/node_modules/@sap/cds-lsp/package.json";
     private static final String RELATIVE_MITM_PATH = "cds-lsp/mitm.js";
     private static final String RELATIVE_LOG_PATH = "cds-lsp/stdio.json";
 
+    private volatile Boolean started = false;
+
     public ServerDescriptor(@NotNull Project project, @NotNull String presentableName) {
         super(project, presentableName);
     }
 
     @NotNull
     @Override
     public OSProcessHandler startServerProcess() throws ExecutionException {
-        checkNodeVersion();
-
         OSProcessHandler handler = new OSProcessHandler(getCommandLine()) {
             @Override
             protected BaseOutputReader.@NotNull Options readerOptions() {
                 return BaseOutputReader.Options.forMostlySilentProcess();
             }
+            @Override
+            protected void onOSProcessTerminated(int exitCode) {
+                super.onOSProcessTerminated(exitCode);
+                if (started && exitCode != 0) {
+                    showUserError(getProcess());
```

Comment:
Is this the UI thread? If yes, why does `started` to be `volatile`? If no, is `UserError.show(...)` synchronizing into the UI therad, then?

Reply:
I just verified in the decompiled sources and at runtime that it's all the same thread, so your comment is perfectly valid. Removing `volatile`.